### PR TITLE
User roles test cases, closes #298

### LINKED ISSUE #298: User roles test cases
Available roles:

- Ozone Secretariat
  - Ozone Secretariat Edit (OS.SE) - can enter data on behalf of a Party (e.g. can create Submissions)
  - Ozone Secretariat Read-Only(OS.SRO) - can view all data reported by any Party, but can't make changes.
- Administrator (OS.A) - view change log for any submission
- Party:
  - Party Reporter (PARTY.PR) - can create Submissons, can view all data reported by his Party. Each Party Reporter is assigned to a single Party. There can be one or more Party Reporters for one Party.
  - Party Read-Only (PARTY.PRO) - It's similar to the Secretariat Read-Only role, but restricted to a single Party.
- Public User (PU) - won't have direct access to any part of the reporting system

Test cases:

1. As OS.SE I can create Submissions on behalf of a Party.
1. As OS.SE I can edit/delete Submissions on behalf of a Party.
1. As OS.SE I can access admin area.
1. As OS.SRO I cannot create/edit/delete Submissions.
1. As OS.SRO I cannot create/edit/delete blends.
1. As OS.SE I can access admin area in read-only mode.
1. As PARTY.PR I can view only data reported by my Party.
1. As PARTY.PR I can only edit/delete Submissions on behalf of my Party.
1. As PARTY.PR I cannot access admin area.
1. PARTY.PRO I can view only data reported by my Party.
1. As PARTY.PR I cannot access admin area.
1. As PARTY.PRO I cannot edit/delete Submissions on behalf of my Party.
1. As PARTY.PRO I cannot create/edit/delete blends on behalf of my Party.
1. As PARTY.PRO I cannot access admin area.
1. As PU I cannot view any Submissions.
1. As PU I cannot create/edit/delete any Submissions.
1. As PU I cannot create/edit/delete any blends.
1. As PU I cannot access admin area.

### PR DESCRIPTION
**Pull request by catalinjitea**
This change is [Reviewable](https://reviewable.io/reviews/eaudeweb/ozone/360)

### CODE REVIEW

File: ozone/tests/factories.py [hunk 1]
```diff
@@ -53,6 +58,9 @@ class Meta:
 
 class SecretariatUserFactory(DjangoModelFactory):
     is_secretariat = True
+    is_superuser = True
```
Comment (sorinstelian) [inline ozone/tests/factories.py]: I don't think any secretariat user is automatically a superuser. Not sure though.

Comment (catalinjitea) [reply ozone/tests/factories.py]: As far as I understood from @nico4 a secretariat user will also be a superuser.

File: ozone/tests/test_clone_permissions.py [hunk 2]
```diff
@@ -71,11 +62,15 @@ def test_clone_secretariat(self):
         )
         submission._current_state = 'submitted'
         submission.save()
-        headers = self.get_authorization_header(self.secretariat_user.username, 'qwe123qwe')
+
```
Comment (sorinstelian) [inline ozone/tests/test_clone_permissions.py]: We should really find a way to avoid all this repeating boilerplate code inside the tests.

Comment (catalinjitea) [reply ozone/tests/test_clone_permissions.py]: We will use `login()` method to authenticate the user. This is a similar approach but using significantly less code.

File: ozone/tests/test_clone_permissions.py [hunk 3]
```diff
@@ -1,8 +1,8 @@
 from django.urls import reverse
```
Comment (sorinstelian) [inline ozone/tests/test_clone_permissions.py]: This file seems to still use the old authentication code.

Comment (catalinjitea) [reply ozone/tests/test_clone_permissions.py]: Nice catch.

**Pull request merged**